app/utils/snack_bar/index.ts

```diff
@@ -12,6 +12,9 @@ export type ShowSnackBarArgs = {
     onAction?: () => void;
     sourceScreen?: AvailableScreens;
     messageValues?: Record<string, PrimitiveType>;
+    autoClose?: boolean; // default is true
+    customMessage?: string;
+    type?: SnackBarConfig['type'];
```

Comment:
Would this be better to directly go for the main source of the type rather than referencing a type of a type?

Or create a MessageType type:

```
export type MessageType = typeof MESSAGE_TYPE[keyof typeof MESSAGE_TYPE];
```

```suggestion
    type?: MessageType;
```

Reply:
Both work exactly the same and not going directly offers some minor benefits of related things automatically updating when changed. I see some benefits of type coupling, but no benefit of not doing it and even some drawback of not doing so.